engine/core/src/main/java/org/pentaho/reporting/engine/classic/core/sorting/SortingDataFactoryCore.java

```diff
@@ -32,7 +32,7 @@ public SortingDataFactoryCore() {
   public String[] getReferencedFields( final DataFactoryMetaData metaData, final DataFactory element,
       final String query, final DataRow parameter ) {
     String[] referencedFields = super.getReferencedFields( metaData, element, query, parameter );
-    if ( referencedFields == null ) {
+    if ( referencedFields == null || referencedFields.length == 0 ) {
```

Comment:
No. The sorting datafactory is a wrapper/decorator around other data factories and you just reverted the good change you made in the table-datafactory by redeclaring the fields as "don't know" instead of 'know for sure I have no fields'. Revert that.

Reply:
Ok, just added because some assert on unit tests are failing due to this fact. Should I change test asserts instead?